libraries/AP_Motors/AP_MotorsHeli_Dual.cpp

```diff
@@ -302,81 +302,61 @@ void AP_MotorsHeli_Dual::calculate_scalars()
-            } else if (swash_num == 2) {
-                swash_tilt = pitch_input + _yaw_scaler * yaw_input;
-            }
-        } else if (swash_axis == AP_MOTORS_HELI_DUAL_SWASH_AXIS_COLL) {
-        // collective
-            if (swash_num == 1) {
-                swash_tilt = 0.45f * _dcp_scaler * (roll_input + constrain_float(_dcp_trim, -0.2f, 0.2f)) + coll_input;
-            } else if (swash_num == 2) {
-                swash_tilt = -0.45f * _dcp_scaler * (roll_input + constrain_float(_dcp_trim, -0.2f, 0.2f)) + coll_input;
-            }
-        }
-    } else if (_dual_mode == AP_MOTORS_HELI_DUAL_MODE_INTERMESHING) {
-        // roll tilt
-        if (swash_axis == AP_MOTORS_HELI_DUAL_SWASH_AXIS_ROLL) {
-            if (swash_num == 1) {
-                swash_tilt = roll_input;
-            } else if (swash_num == 2) {
-                swash_tilt = roll_input;
-            }
-        } else if (swash_axis == AP_MOTORS_HELI_DUAL_SWASH_AXIS_PITCH) {
-        // pitch tilt
-            if (swash_num == 1) {
-                swash_tilt = pitch_input - _yaw_scaler * yaw_input;
-            } else if (swash_num == 2) {
-                swash_tilt = pitch_input + _yaw_scaler * yaw_input;
-            }
-        } else if (swash_axis == AP_MOTORS_HELI_DUAL_SWASH_AXIS_COLL) {
-        // collective
-            if (swash_num == 1) {
-                swash_tilt = 0.45f * _dcp_scaler * yaw_input + coll_input;
-            } else if (swash_num == 2) {
-                swash_tilt = -0.45f * _dcp_scaler * yaw_input + coll_input;
-            }
-        }
-    } else { // AP_MOTORS_HELI_DUAL_MODE_TANDEM
-        // roll tilt
-        if (swash_axis == AP_MOTORS_HELI_DUAL_SWASH_AXIS_ROLL) {
-            if (swash_num == 1) {
-                swash_tilt = roll_input + _yaw_scaler * yaw_input;
-            } else if (swash_num == 2) {
-                swash_tilt = roll_input - _yaw_scaler * yaw_input;
-            }
-        } else if (swash_axis == AP_MOTORS_HELI_DUAL_SWASH_AXIS_PITCH) {
-        // pitch tilt
-            if (swash_num == 1) {
-                swash_tilt = 0.0f;
-            } else if (swash_num == 2) {
-                swash_tilt = 0.0f;
-            }
-        } else if (swash_axis == AP_MOTORS_HELI_DUAL_SWASH_AXIS_COLL) {
-        // collective
-            if (swash_num == 1) {
-                swash_tilt = 0.45f * _dcp_scaler * (pitch_input + constrain_float(_dcp_trim, -0.2f, 0.2f)) + coll_input;
-            } else if (swash_num == 2) {
-                swash_tilt = -0.45f * _dcp_scaler * (pitch_input + constrain_float(_dcp_trim, -0.2f, 0.2f)) + coll_input;
-            }
-        }
-    }
-    return swash_tilt;
+    // Differential cyclic roll is used for yaw and combined for roll
+    const float swash1_roll = roll_input + _yaw_scaler * yaw_input;
+    const float swash2_roll = roll_input - _yaw_scaler * yaw_input;
+
+    // cyclic is not used for pitch control
+    const float pitch = 0.0;
```

Comment:
nit pick: can we call this something like swash_pitch, just for readability.

Reply:
done